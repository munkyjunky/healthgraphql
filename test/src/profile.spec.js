const nock = require('nock');
const constants = require('../../src/constants/healthgraph');
const healthgraphql = require('../helpers/healthgraphql');
const root = require('../res/root');
const profileResponse = require('../res/profile');


describe('Profile', () => {

    beforeEach(() => {
        // mock API HTTP response
        nock(constants.BASE_URL)
            .get(constants.ROOT)
            .reply(200, root);

        nock(constants.BASE_URL)
            .get(root.profile)
            .reply(200, profileResponse);
    });

    it('should return the correct userID', () => {

        const query = `{ userID }`;

        return healthgraphql(query).then(res => {
            expect(res).toMatchSnapshot();
        });

    });

    it('should return correct profile information', () => {

        const query = `{
            profile {
                name,
                location,
                birthday
            } 
        }`;

        return healthgraphql(query).then(res => {
            expect(res).toMatchSnapshot();
        });

    });

});
