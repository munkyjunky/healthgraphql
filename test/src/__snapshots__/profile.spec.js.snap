// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should return correct profile information 1`] = `
Object {
  "data": Object {
    "profile": Object {
      "birthday": "Sat, 1 Jan 2011 00:00:00",
      "location": "London, UK",
      "name": "Test User",
    },
  },
}
`;

exports[`Profile should return the correct userID 1`] = `
Object {
  "data": Object {
    "userID": "1234567890",
  },
}
`;
